openedx/core/djangoapps/content_libraries/tests/test_runtime.py

```diff
@@ -182,6 +186,43 @@ def test_xblock_metadata(self):
         assert metadata_view_result.data['student_view_data'] is None
         # Capa doesn't provide student_view_data
 
+    @skip_unless_cms  # modifying blocks only works properly in Studio
+    def test_xblock_fields(self):
+        """
+        Test the XBlock fields API
+        """
+        # act as staff:
+        client = APIClient()
+        client.login(username=self.staff_user.username, password='edx')
+
+        # create/save a block using the library APIs first
+        unit_block_key = library_api.create_library_block(self.library.key, "unit", "fields-u1").usage_key
+        block_key = library_api.create_library_block_child(unit_block_key, "html", "fields-p1").usage_key
+        new_olx = """
+        <html display_name="New Text Block">
+            <p>This is some <strong>HTML</strong>.</p>
+        </html>
+        """.strip()
+        library_api.set_library_block_olx(block_key, new_olx)
+        library_api.publish_changes(self.library.key)
+
+        # Check the GET API for the block:
+        fields_get_result = client.get(URL_BLOCK_FIELDS_URL.format(block_key=block_key))
+        assert fields_get_result.data['display_name'] == 'New Text Block'
+        assert fields_get_result.data['data'].strip() == '<p>This is some <strong>HTML</strong>.</p>'
+        assert fields_get_result.data['metadata']['display_name'] == 'New Text Block'
+
+        # Check the POST API for the block:
+        fields_post_result = client.post(URL_BLOCK_FIELDS_URL.format(block_key=block_key), data={
+            'data': '<p>test</p>',
+            'metadata': {
+                'display_name': 'New Display Name',
+            }
+        }, format='json')
+        block_saved = xblock_api.load_block(block_key, self.staff_user)
+        assert block_saved.data == '\n<p>test</p>\n'
```

Comment:
Will it be an issue for the editors if some newlines go inserted here? This test _set_ the data to `'<p>test</p>'` but we _get_ the data back as `'\n<p>test</p>\n'`. I'm assuming that's fine but just want to confirm.

Reply:
no, in fact I don't think editors can have content without the newlines; it's just an artifact of what I wrote in the test.